DUI3-DX/Converters/ArcGIS/Speckle.Converters.ArcGIS3/Features/GisFeatureToSpeckleConverter.cs

```diff
@@ -2,41 +2,99 @@
 
   public GisFeature RawConvert(Row target)
   {
-    var shape = (ACG.Geometry)target["Shape"];
-    var speckleShapes = _geometryConverter.RawConvert(shape).ToList();
-
     // get attributes
     var attributes = new Base();
     IReadOnlyList<Field> fields = target.GetFields();
-    int i = 0;
     foreach (Field field in fields)
     {
       string name = field.Name;
+      if (name != "Shape") // ignore the field with geometry itself
+      {
+        try
+        {
+          var value = target[name];
+          attributes[name] = value; // can be null
+        }
+        catch (ArgumentException)
+        {
+          // TODO: log in the conversion errors list
+          attributes[name] = null;
+        }
+      }
+    }
+
+    try
+    {
+      var shape = (ACG.Geometry)target["Shape"];
+      var speckleShapes = _geometryConverter.RawConvert(shape).ToList();
+
+      // re-shape the GisFeature:
+      // if shapes is a list of Meshes, set them as DisplayValue instead of geometry
+      if (speckleShapes.Count > 0 && speckleShapes[0] is SOG.Mesh)
+      {
+        var displayVal = speckleShapes;
+        return new GisFeature(attributes, displayVal);
+      }
 
-      // breaks on Raster Field type
-      if (name != "Shape" && field.FieldType.ToString() != "Raster")
+      // if shapes is a list of GisPolygonGeometry, create DisplayValue for those with no voids
+      if (speckleShapes.Count > 0 && speckleShapes[0] is GisPolygonGeometry)
       {
-        var value = target.GetOriginalValue(i); // can be null
-        attributes[name] = value;
+        List<Base> displayVal = new();
+        foreach (var poly in speckleShapes)
+        {
+          if (poly is GisPolygonGeometry polygon && polygon.voids.Count == 0)
+          {
+            // counter-clockwise orientation for up-facing mesh faces
+            List<SOG.Point> boundaryPts = polygon.boundary.GetPoints();
+            bool isClockwise = _geomUtils.IsClockwisePolygon(boundaryPts);
+            if (isClockwise)
+            {
+              boundaryPts.Reverse();
+            }
+
+            // generate Mesh
+            int ptCount = boundaryPts.Count;
+            List<int> faces = new() { ptCount };
+            faces.AddRange(Enumerable.Range(0, ptCount).ToList());
+
+            displayVal.Add(
+              new SOG.Mesh(boundaryPts.SelectMany(x => new List<double> { x.x, x.y, x.z }).ToList(), faces)
+            );
+          }
+        }
+        // POC: how to deal with multiploygons, where not all would have displayValue? Will they not be all received in other apps?
+        return new GisFeature(speckleShapes, attributes, displayVal);
       }
-      i++;
+
+      // otherwise set shapes as Geometries
+      return new GisFeature(speckleShapes, attributes);
+    }
+    catch (Exception ex) when (ex is ArgumentOutOfRangeException or GeodatabaseFieldException) // if no geometry
+    {
+      return new GisFeature(attributes);
     }
```

Comment:
I'm leaning towards not accepting this. If this exception is trying to catch the "no geometry" scenario, I'm sure there's a better way to do it than letting it explode.

This also doesn't let us know if the problem was that the ArcGIS feature had no geometry, if we failed to convert it, or if we failed to mesh the geometry.

These are 3 separate edge cases that ideally should be dealt with separately (and try/catched with whatever exceptions we can recover from.

Reply:
These exceptions are catching a specific case when the field "Shape" in inaccessible, but fair point, changed it to IF statement 👍